homeassistant/components/enigma2/media_player.py

```diff
@@ -80,49 +71,67 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up of an enigma2 media player."""
-    if discovery_info:
-        # Discovery gives us the streaming service port (8001)
-        # which is not useful as OpenWebif never runs on that port.
-        # So use the default port instead.
-        config[CONF_PORT] = DEFAULT_PORT
-        config[CONF_NAME] = discovery_info["hostname"]
-        config[CONF_HOST] = discovery_info["host"]
-        config[CONF_USERNAME] = DEFAULT_USERNAME
-        config[CONF_PASSWORD] = DEFAULT_PASSWORD
-        config[CONF_SSL] = DEFAULT_SSL
-        config[CONF_USE_CHANNEL_ICON] = DEFAULT_USE_CHANNEL_ICON
-        config[CONF_MAC_ADDRESS] = DEFAULT_MAC_ADDRESS
-        config[CONF_DEEP_STANDBY] = DEFAULT_DEEP_STANDBY
-        config[CONF_SOURCE_BOUQUET] = DEFAULT_SOURCE_BOUQUET
-
-    base_url = URL.build(
-        scheme="https" if config[CONF_SSL] else "http",
-        host=config[CONF_HOST],
-        port=config.get(CONF_PORT),
-        user=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
+
+    host = config[CONF_HOST]
+
+    async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2024.7.0",
+        is_fixable=False,
+        is_persistent=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Enigma2",
+        },
     )
 
-    session = async_create_clientsession(hass, verify_ssl=False, base_url=base_url)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.options[CONF_HOST] == host:
+            return
 
-    device = OpenWebIfDevice(
-        host=session,
-        turn_off_to_deep=config.get(CONF_DEEP_STANDBY, False),
-        source_bouquet=config.get(CONF_SOURCE_BOUQUET),
+    entry_data = {
+        CONF_NAME: config.get(CONF_NAME),
```

Comment:
Please don't use `dict.get` for keys which are guaranteed to be in the config

Reply:
In the YAML config, only the host is required, everything else is optional.

Reply:
That's because you modified the platform schema to remove the defaults, please revert that change.